c10/core/Allocator.h

```diff
@@ -115,8 +116,10 @@ class C10_API DataPtr {
   }
   // Unsafely mutates the device on a DataPtr.  Under normal use,
   // you should never actually need to call this function.
-  // We need this for the implementation of the hack detailed
-  // in Note [Masquerading as CUDA]
+  // We used to need this for the implementation of the hack detailed
+  // in Note [Masquerading as CUDA], but that hack has been removed.
+  C10_DEPRECATED_MESSAGE(
```

Comment:
Why deprecate this?
It does seem to be used outside of core: https://github.com/search?q=%2Funsafe_set_device%5C%28%2F+language%3AC%2B%2B&type=code&l=C%2B%2B

Reply:
Awesome!  I didn't realize.  I will remove the deprecation.  I think this function was first created for this nonsense about caffe2 has a HIP device but torch only has the HIP-as-CUDA device.  Good catch here, thanks.